code/datums/martial/boxing.dm

```diff
@@ -1,102 +1,273 @@
-	return MARTIAL_ATTACK_SUCCESS
+	
+	if(grant_experience)
+		skill_experience_adjustment(attacker, (damage/lower_force))
 
-/datum/martial_art/boxing/can_use(mob/living/martial_artist)
-	if(!ishuman(martial_artist))
+	if(defender.stat == DEAD || !honor_check(defender)) //early returning here so we don't worry about knockout probs
+		return TRUE
+
+	//Determine our attackers athletics level as a knockout probability bonus
+	var/attacker_athletics_skill =  (attacker.mind?.get_skill_modifier(/datum/skill/athletics, SKILL_RANDS_MODIFIER) + base_unarmed_effectiveness)
+		
+	// Defender boxing skill and armor block are used as a defense here. This has already factored in base_unarmed_effectiveness from the attacker
+	var/defender_athletics_skill =  clamp(defender.mind?.get_skill_modifier(/datum/skill/athletics, SKILL_RANDS_MODIFIER), 0, 100)
+		
+	//Determine our final probability, using a clamp to stop any prob() weirdness.
+	var/final_knockout_probability = clamp(round(attacker_athletics_skill - defender_athletics_skill), 0 , 100)
+	
+	if(!prob(final_knockout_probability))
+		return TRUE
+	
+	if(defender.get_timed_status_effect_duration(/datum/status_effect/staggered))
+		defender.visible_message(
+			span_danger("[attacker] knocks [defender] out with a haymaker!"),
+			span_userdanger("You're knocked unconscious by [attacker]!"),
+			span_hear("You hear a sickening sound of flesh hitting flesh!"),
+			COMBAT_MESSAGE_RANGE,
+			attacker,
+		)
+		to_chat(attacker, span_danger("You knock [defender] out with a haymaker!"))
+		defender.apply_effect(20 SECONDS, EFFECT_KNOCKDOWN, armor_block)
+		defender.SetSleeping(10 SECONDS)
+		log_combat(attacker, defender, "knocked out (boxing) ")
+	else
+		defender.visible_message(
+			span_danger("[attacker] staggers [defender] with a haymaker!"),
+			span_userdanger("You're nearly knocked off your feet by [attacker]!"),
+			span_hear("You hear a sickening sound of flesh hitting flesh!"),
+			COMBAT_MESSAGE_RANGE,
+			attacker,
+		)
+		defender.adjust_staggered_up_to(STAGGERED_SLOWDOWN_LENGTH, 10 SECONDS)
+		to_chat(attacker, span_danger("You stagger [defender] with a haymaker!"))
+		log_combat(attacker, defender, "staggered (boxing) ")
+	
+	playsound(defender, 'sound/effects/coin2.ogg', 40, TRUE)
+	new /obj/effect/temp_visual/crit(get_turf(defender))
+	skill_experience_adjustment(attacker, (damage/lower_force)) //double experience for a successful crit
+	
+	return TRUE
+
+/// Returns whether whoever is checked by this proc is complying with the rules of boxing. The boxer cannot block non-boxers, and cannot apply their scariest moves against non-boxers.
+/datum/martial_art/boxing/proc/honor_check(mob/living/possible_boxer)
+	if(!honorable_boxer)
+		return TRUE //You scoundrel!!
+	
+	if(!ishuman(possible_boxer))
 		return FALSE
-	return ..()
 
-/obj/item/clothing/gloves/boxing
-	var/datum/martial_art/boxing/style
+	if(HAS_TRAIT(possible_boxer, TRAIT_BOXING_READY))
+		return FALSE
+	
+	return TRUE
```

Comment:
Unless I am reading it wrong, isn't this logic inverted?
You could also potentially skip the ishuman check and only look for the trait in order to introduce the possibility of boxing mobs but that's probably a silly concept.

Reply:
Oh yes that is inverted i must have missed a ! while changing it to a trait check